src/libfsm/intersect.c

```diff
@@ -6,13 +6,15 @@
 
 #include <assert.h>
 #include <stddef.h>
-#include <limits.h>
 #include <errno.h>
 
 #include <fsm/fsm.h>
+#include <fsm/out.h>
+#include <fsm/options.h>
 #include <fsm/bool.h>
 
 #include "internal.h"
+#include "walk2.h"
```

Comment:
I'd rather no change to this file on this PR; same for subtract.c.

Reply:
The intermediate changesets involve quite a few changes to intersect.c.  Should I squash them into one changeset or are intermediates okay as long as the PR as a whole leaves intersect.c (and others) untouched?

Reply:
Whatever works for you. I think if it were me, I'd remove everything from the history which edited this file (by `git rebase -i` and editing/dropping those commits). But I don't mind.

Reply:
The changes are quite extensive; fsm_walk2 grew out of the intersection-by-dfs code in intersect.c

I should probably squash the commits so it's more sensible to a reader that doesn't know the sordid history. :)